tracer/test/test-applications/instrumentation/CallTargetNativeTest/NoOp/RefStructTwoParametersVoidIntegration.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace CallTargetNativeTest.NoOp;
+
+public static class RefStructTwoParametersVoidIntegration
+{
+    public static CallTargetState OnMethodBegin<TTarget, TArg1, TArg2>(TTarget instance, ref TArg1 arg01, ref TArg2 arg02)
+    {
+        var returnValue = CallTargetState.GetDefault();
+        Console.WriteLine($"ProfilerOK: BeginMethod(2)<{typeof(RefStructTwoParametersVoidIntegration)}, {typeof(TTarget)}, {typeof(TArg1)}, {typeof(TArg2)}>({instance}, {arg01}, {arg02})");
+
+        bool success;
+        if (arg01 is CallTargetRefStruct callTargetRefStruct)
+        {
+            ref var readOnlySpanValue = ref callTargetRefStruct.GetReadOnlySpan<char>(out success);
+            if (success)
+            {
+                readOnlySpanValue = "Hello";
```

Comment:
We should never reassign the span (I wonder if we can declare the argument as `ref readonly` to prevent that). Part of the caller code could rely on the size of the Span.

```csharp
readOnlySpanValue[0] = 'a'; // fine
readOnlySpanValue = new ReadOnlySpan<>(...); // dangerous
```

Reply:
Actually I don't think we can force the argument of the calltarget method to be `ref readonly`, but we can do it for the return value of `callTargetRefStruct.GetReadOnlySpan`

Reply:
I've changed the return value of the `Get..` and added a `DangerousGet...` for those cases were we really want to modify the span. Changes are in [bb59282](https://github.com/DataDog/dd-trace-dotnet/pull/5442/commits/bb59282665a0d799431a31a11a6bdd5d9c477a09)